packages/gatsby/src/schema/types/pagination.js

```diff
@@ -18,26 +18,34 @@ const getEdge = ({ schemaComposer, typeComposer }) => {
   return schemaComposer.getOrCreateTC(typeName, tc => {
     tc.addFields({
       next: typeComposer,
-      node: typeComposer,
+      node: typeComposer.getTypeNonNull(),
       previous: typeComposer,
     })
   })
 }
 
-const createPagination = ({ schemaComposer, typeComposer, fields, typeName }) =>
-  schemaComposer.getOrCreateTC(typeName, tc => {
+const createPagination = ({
+  schemaComposer,
+  typeComposer,
+  fields,
+  typeName,
+}) => {
+  const paginationTypeComposer = schemaComposer.getOrCreateTC(typeName, tc => {
     tc.addFields({
-      totalCount: `Int`,
-      edges: [getEdge({ schemaComposer, typeComposer })],
-      pageInfo: getPageInfo({ schemaComposer }),
+      totalCount: `Int!`,
+      edges: [getEdge({ schemaComposer, typeComposer }).getTypeNonNull()],
```

Comment:
Array itself should also be non-null.

Reply:
While we are at it, could you add `nodes` field as a shortcut to `edges { node }`.

Reply:
`edges` array is made nonnull further down with `makeFieldNonNull(`edges`)`.

Reply:
shortcut field: 100% yes! should be a new PR?

Reply:
Can be the same.